pkg/ccl/importccl/import_stmt_test.go

```diff
@@ -4540,3 +4542,212 @@ func TestDisallowsInvalidFormatOptions(t *testing.T) {
 		}
 	}
 }
+
+var importSafeFunctions = map[string]struct{}{
+	"current_timestamp": {},
+	"now":               {},
+	"localtimestamp":    {},
+}
+
+// importSafeEvalVisitor is a visitor which verifies if
+// the expression is safe to be used in default expression during import.
+// The returned expression from tree.WalkExpr() is an evaluated
+// expression of appropriate type.
+// NB: This visitor must be invoked only on tree.TypedExpr expressions.
+type importSafeEvalVisitor struct {
+	ctx *tree.EvalContext
+	err error
+}
+
+var _ tree.Visitor = &importSafeEvalVisitor{}
+
+func (v *importSafeEvalVisitor) VisitPre(expr tree.Expr) (bool, tree.Expr) {
+	return v.err == nil, expr
+}
+
+func (v *importSafeEvalVisitor) VisitPost(expr tree.Expr) tree.Expr {
+	// Since we require that the expression we begin with is a tree.TypedExpr,
+	// we no longer have to have these checks here.  We can safely cast
+	// expressions to their typed variants.
+	if fn, ok := expr.(*tree.FuncExpr); ok {
+		resolvedFnName := fn.Func.FunctionReference.(*tree.FunctionDefinition).Name
+		if _, ok := importSafeFunctions[resolvedFnName]; !ok {
+			v.err = errors.Newf("function %q not safe for import", tree.AsString(fn))
+		}
+	}
+
+	if v.err != nil {
+		return expr
+	}
+
+	value, err := expr.(tree.TypedExpr).Eval(v.ctx)
+	if err != nil {
+		v.err = err
+		return expr
+	}
+	return value
+}
+
+func unsafeExpressionError(err error, expr tree.Expr) error {
+	return errors.Wrapf(err,
+		"default expression %q not safe for import", tree.AsString(expr))
+}
+
+func SanitizeExpressionForImport(
+	ctx context.Context, evalCtx *tree.EvalContext, expr tree.Expr, targetType *types.T,
+) (tree.Expr, error) {
+	semaCtx := tree.MakeSemaContext()
+
+	// Immutable expressions are fine.
+	typedExpr, err := sqlbase.SanitizeVarFreeExpr(
+		ctx, expr, targetType, "isImportSafe", &semaCtx, tree.VolatilityImmutable)
+	if err == nil {
+		return typedExpr.Eval(evalCtx)
+	}
+
+	// We know we're dealing with tree.VolatilityStable or more volatile.
+	// Still, the result of this expression must be correct type.
+	typedExpr, err = expr.TypeCheck(ctx, &semaCtx, targetType)
+	if err != nil {
+		return nil, unsafeExpressionError(err, expr)
+	}
+
+	// Now that we have typed expression, walk it.
+	v := &importSafeEvalVisitor{ctx: evalCtx}
+	newExpr, _ := tree.WalkExpr(v, typedExpr)
+	if v.err != nil {
+		return nil, unsafeExpressionError(v.err, expr)
+	}
+
+	// Evaluate result expression.
+	typedExpr, err = newExpr.TypeCheck(ctx, &semaCtx, targetType)
```

Comment:
I don't think you need to typecheck this again, walking a typedexpr with your visitor should return a typed expr